### PR TITLE
Preload the edited template to avoid the white page effect before the screen loads

### PR DESCRIPTION
**Pull request by youknowriad**
Right now the screen loads as a white page before the API fetching the template finishes. This PR fixes that by preloading the template.

### CODE REVIEW

**Comment by mtias** [comment]
Nice!

File: lib/edit-site-page.php [hunk 1]
```diff
@@ -208,6 +208,28 @@ function gutenberg_edit_site_init( $hook ) {
...
 
+	// Preload block editor paths.
+	// most of these are copied from edit-forms-blocks.php.
+	$preload_paths = array(
+		'/',
+		'/wp/v2/types?context=edit',
+		'/wp/v2/taxonomies?per_page=-1&context=edit',
+		'/wp/v2/themes?status=active',
+		sprintf( '/wp/v2/templates/%s?context=edit', $_wp_current_template_id ),
+		'/wp/v2/users/me?post_type=templates&context=edit',
```
Comment (Addison-Stavlo) [inline lib/edit-site-page.php]: does having `post_type=templates` make sense here?  For the site editor isn't our post type `undefined` ?

**Review by Addison-Stavlo — COMMENTED** [review]
This works well!  Very nice.

I see from the comment the preload paths are taken from `edit-forms-blocks.php.` Do all of these items in the path make sense to have preloaded for the site editor?  By the looks of it most of them seem appropriate, but Im not certain.

Anyways, testing this change I do see a noticeable difference!

File: lib/edit-site-page.php (hunk 1 above)
Comment (youknowriad) [reply lib/edit-site-page.php]: I think our post type is wp_template actually, so yeah that might be the only unused entry here, I'll test it.

File: packages/block-editor/src/hooks/color.js [hunk 2]
```diff
@@ -139,6 +139,9 @@ const ColorPanel = ( { colorSettings, clientId } ) => {
 
 	useEffect( () => {
 		const colorsDetectionElement = getBlockDOMNode( clientId );
+		if ( ! colorsDetectionElement ) {
```
Comment (youknowriad) [inline packages/block-editor/src/hooks/color.js]: This is a small unrelated bug fix that was causing some blocks (template part) to fail in the edit site page.

**Review by Addison-Stavlo — APPROVED** [review]
Works well & looks good from my end!
